arrows/vxl/tests/test_vidl_ffmpeg_video_input.cxx

```diff
@@ -0,0 +1,391 @@
+}
+
+// ----------------------------------------------------------------------------
+TEST_F(vidl_ffmpeg_video_input, is_good)
+{
+  // make config block
+  auto config = kwiver::vital::config_block::empty_config();
+
+  kwiver::arrows::vxl::vidl_ffmpeg_video_input vfvi;
+
+  vfvi.check_configuration( config );
+  vfvi.set_configuration( config );
+
+  kwiver::vital::path_t video_file = data_dir + "/" + video_file_name;
+  kwiver::vital::timestamp ts;
+
+  EXPECT_FALSE( vfvi.good() )
+    << "Video state before open";
+
+  // open the video
+  vfvi.open( video_file );
+  EXPECT_FALSE( vfvi.good() )
+    << "Video state after open but before first frame";
+
+  // step one frame
+  vfvi.next_frame( ts );
+  EXPECT_TRUE( vfvi.good() )
+    << "Video state on first frame";
+
+  // close the video
+  vfvi.close();
+  EXPECT_FALSE( vfvi.good() )
+    << "Video state after close";
+
+  // Reopen the video
+  vfvi.open( video_file );
+
+  int num_frames = 0;
+  while ( vfvi.next_frame( ts ) )
+  {
+    ++num_frames;
+    EXPECT_TRUE( vfvi.good() )
+      << "Video state on frame " << ts.get_frame();
+  }
+  EXPECT_EQ( num_expected_frames, num_frames );
+  EXPECT_EQ( num_expected_frames, vfvi.num_frames() )
+    << "Number of frames after checking frames should be "
+    << num_expected_frames;
+}
+
+TEST_F(vidl_ffmpeg_video_input, seek_frame)
+{
+  // make config block
+  auto config = kwiver::vital::config_block::empty_config();
+
+  kwiver::arrows::vxl::vidl_ffmpeg_video_input vfvi;
+
+  vfvi.check_configuration( config );
+  vfvi.set_configuration( config );
+
+  kwiver::vital::path_t video_file = data_dir + "/" + video_file_name;
+
+  // Open the video
+  vfvi.open( video_file );
+
+  test_seek_frame( vfvi );
+
+  vfvi.close();
+}
+
+TEST_F(vidl_ffmpeg_video_input, seek_frame_sublist)
+{
+  // make config block
+  auto config = kwiver::vital::config_block::empty_config();
+
+  config->set_value( "start_at_frame", "11" );
+  config->set_value( "stop_after_frame", "30" );
+
+  kwiver::arrows::vxl::vidl_ffmpeg_video_input vfvi;
+
+  vfvi.check_configuration( config );
+  vfvi.set_configuration( config );
+
+  kwiver::vital::path_t video_file = data_dir + "/" + video_file_name;
+
+  // Open the video
+  vfvi.open( video_file );
+
+  test_seek_frame_sublist( vfvi );
+
+  vfvi.close();
+}
+
+TEST_F(vidl_ffmpeg_video_input, metadata_map)
+{
+  // make config block
+  auto config = kwiver::vital::config_block::empty_config();
+
+  kwiver::arrows::vxl::vidl_ffmpeg_video_input vfvi;
+
+  vfvi.check_configuration( config );
+  vfvi.set_configuration( config );
+
+  kwiver::vital::path_t video_file = data_dir + "/" + video_file_name;
+
+  // Open the video
+  vfvi.open( video_file );
+
+  // Get metadata map
+  auto md_map = vfvi.metadata_map()->metadata();
+
+  // Each frame of video should have some metadata
+  // at a minimum this is just the video name and timestamp
+  EXPECT_EQ( md_map.size(), vfvi.num_frames() );
```

Comment:
@chetnieter  This is the one test on this branch that is failing (not counting what is also failing on master).  Previously it was comparing `md_map.size()` to 0; however, there was a change in the vidl_ffmpeg_video_input in that it now creates a basic metadata object for every frame.  That basic object contains the video filename and frame number if no other info is available.  So I changed this test to compare to `vfvi.num_frames()`.  It still fails because `md_map.size() == 51` and `vfvi.num_frames() == 50`.  I added code below to print out the frame numbers in the map and found 0-50 instead of 1-50.  There seems to be a bug.  Can you look into this?

Reply:
@chetnieter I've dug into this a bit myself and submitted several fixes and added additional test cases for building the metadata map with a subset of frames, skipping frames, and not calling this function on the first frame.  These tests exposed a few new issues.  I had to rework how the frame skipping was done to make it more consistent.

All tests pass now, but you might wish to review my changes.

Reply:
@mleotta I looked over your last four commits and things look good to me. Does the bug involving the difference in `md_map.size()` and `vfvi.num_frames()` still need to be resolved?

Reply:
No that was addressed as well